scrapy_po/utils.py

```diff
@@ -129,3 +133,26 @@ def build_instances(plan: andi.Plan, providers):
         else:
             instances[cls] = cls(**kwargs_spec.kwargs(instances))
     raise returnValue(instances)
+
+
+def callback_for(page_cls: Type[ItemPage]) -> Callable:
+    """Helper for creating callbacks for ItemPage sub-classes."""
+    if not issubclass(page_cls, ItemPage):
+        raise TypeError(
+            f'{page_cls.__name__} should be a sub-class of ItemPage.')
+
+    if getattr(page_cls.to_item, '__isabstractmethod__', False):
+        raise NotImplementedError(
+            f'{page_cls.__name__} should implement to_item method.')
+
+    # When the callback is used as an instance method of the spider, it expects
+    # to receive 'self' as its first argument. When used as a simple inline
+    # function, it expects to receive a response as its first argument.
+    #
+    # To avoid a TypeError, we need to receive a list of unnamed arguments and
+    # a dict of named arguments after our injectable.
+    def parse(*args, page: page_cls, **kwargs):  # type: ignore
+        yield page.to_item()  # type: ignore
```

Comment:
`return` instead of `yield`?

Reply:
Keeping old behavior... But I've had some experiences when the spider ignored items "returned" while processed items "yielded". Possibly related to async behavior (inline callbacks)?